### DISCUSSION TITLE
Limit upload size by volume

### DISCUSSION THREAD

**Discussion by rachelrine** [craftcms/cms#8613] [Ideas]
Would be nice to have file upload limits specific to asset volumes. Making the global upload limit large enough to accommodate large PDF files means also allowing larger image uploads.

**Comment by turnstylerj** [comment]
This still ends up being an issue on almost every one of our builds. Would love to see this addressed.

**Comment by timkelty** [comment]
Ideally, I think we'd:

- add file size as a restriction on assets fields
- add ALL restrictions (include file size) that exist on fields as volume settings.

Eventually it may make sense to even remove these settings from fields, as it would likely always make more sense to configure these at the volume level,

**Reply by turnstylerj** [reply]
Yeah, either of these would be super helpful and solve the issue for us.

**Comment by piotrpog** [comment]
Definitely something that should be introduced.
